llvm/lib/Target/AArch64/AsmParser/AArch64AsmParser.cpp

```diff
@@ -4546,18 +4548,21 @@ ParseStatus AArch64AsmParser::tryParseZTOperand(OperandVector &Operands) {
     if (getParser().parseExpression(ImmVal))
       return ParseStatus::NoMatch;
     const MCConstantExpr *MCE = dyn_cast<MCConstantExpr>(ImmVal);
+    Operands.push_back(AArch64Operand::CreateImm(
+        MCConstantExpr::create(MCE->getValue(), getContext()), StartLoc,
+        getLoc(), getContext()));
     if (!MCE)
```

Comment:
I think we should change the order :
```
if (!MCE)
  return TokError("immediate value expected for vector index");

    Operands.push_back(AArch64Operand::CreateImm(
        MCConstantExpr::create(MCE->getValue(), getContext()), StartLoc,
        getLoc(), getContext()));

```
Just because it is only acceptable to get the value (MCE->getValue()) and push to the operand if we could actually create the constant.